src/software/ai/hl/stp/play/ball_placement/ball_placement_play_fsm.cpp

```diff
@@ -21,13 +22,69 @@ void BallPlacementPlayFSM::kickOffWall(const Update &event)
 
     // setup wall kickoff tactic for ball placing robot
     Point ball_pos            = event.common.world_ptr->ball().position();
-    Rectangle field_lines     = event.common.world_ptr->field().fieldLines();
-    AutoChipOrKick auto_chick = {AutoChipOrKickMode::AUTOKICK,
-                                 WALL_KICKOFF_VELOCITY_M_PER_S};
+    Rectangle field_boundary     = event.common.world_ptr->field().fieldBoundary();
+
+    pickoff_final_orientation = calculateWallPickOffLocation(ball_pos, field_boundary, ROBOT_MAX_RADIUS_METERS * 4).first;
+    pickoff_point = ball_pos - Vector::createFromAngle(pickoff_final_orientation).normalize(0.4);
+    align_wall_tactic->updateControlParams(pickoff_point, pickoff_final_orientation,
+                                           0.0, TbotsProto::DribblerMode::OFF,
+                                           TbotsProto::BallCollisionType::AVOID, {AutoChipOrKickMode::OFF, 0},
+                                           TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
+                                           TbotsProto::ObstacleAvoidanceMode::SAFE, 0.0);
```

Comment:
should it be `SAFE` or `AGGRESSIVE`?

Reply:
I think `SAFE` is fine. This only affects how we motion plan around enemy robots, and it's probably better to be safe. If this causes an issue, chances are it's the enemy's fault